character.cpp

```diff
@@ -1,17 +1,17 @@
 #include "character.h"
 
 
-Character::Character(const std::string& name, int health, int damage) :name(name), health(health), damage(damage) {}
+
+Character::Character(const std::string& name, int health, const int& damage) :name(name), health(health), damage(damage) {}
 std::string Character::getName() const { return name; }
 int Character::getDamage() const { return damage; }
 int Character::getHealth() const { return health; }
 
 
 
-void Character::healthLoss(const Character* Attacker) {
-	health -= Attacker->damage;
+void Character::reduceHealthByDamage(Character &attacker) {

```

Comment:
De a const miért tűnt el? Ahogy a mutató const volt, úgy a referenciának is constnak kéne lennie.